[openssh] Fix hook/init script to inject full path to ssh-keygen:

gavindidrichsen opened the pull request:
In other words, instead of referencing ssh-keygen use {{pkg.path}}/bin/ssh-keygen.  This fix was initially added as part of https://github.com/habitat-sh/core-plans/pull/2375, but has been pulled out into this its own separate PR.

Signed-off-by: Gavin Didrichsen <gavin.didrichsen@gmail.com>

echohack commented:
Hi @gavindidrichsen 

Can you provide test steps? Here's an example:

##Testing

hab studio enter
./telegraf/tests/test.sh

##Sample output

 ✓ Version matches
 ✓ Service is running

2 tests, 0 failures

echohack commented:
I think this is good to merge, but the `core/git` transitive dep makes me want to have a 2nd approval here, as git is used as a `pkg_build_deps` nearly everywhere. @habitat-sh/habitat-core-plans-maintainers

echohack commented:
@gavindidrichsen Gentle nudge here. Could you describe the issue you encountered and are solving with this?

gavindidrichsen commented:
Closing this PR.  Background to this PR is that this change was 
* originally included as part of another PR [inspec testing the openssh service](https://github.com/habitat-sh/core-plans/pull/2375)
* removed from the original PR and added as its own PR because to separate a fundamental change of the core plan from the addition of tests.

It has been found that this change does not impact the successful build and installation of the openssh service; therefore it is being closed.
